skyportal/models.py

```diff
@@ -1179,4 +1183,45 @@ def set_time(self):
     foreign_keys="ClassicalAssignment.requester_id",
 )
 
+
+class Invitation(Base):
+    token = sa.Column(sa.String(), nullable=False, unique=True)
+    groups = relationship(
+        "Group",
+        secondary="group_invitations",
+        cascade="save-update, merge, refresh-expire, expunge",
+        passive_deletes=True,
+    )
+    admin_for_groups = sa.Column(psql.ARRAY(sa.Boolean), nullable=False)
+    user_email = sa.Column(EmailType(), nullable=True)
+    invited_by = relationship(
+        "User",
+        secondary="user_invitations",
+        cascade="save-update, merge, refresh-expire, expunge",
+        passive_deletes=True,
+        uselist=False,
+    )
+
+
+GroupInvitation = join_model('group_invitations', Group, Invitation)
+UserInvitation = join_model("user_invitations", User, Invitation)
+
+
+@event.listens_for(Invitation, 'after_insert')
+def send_user_invite_email(mapper, connection, target):
+    _, cfg = load_env()
+    message = Mail(
+        from_email=cfg["invitations.from_email"],
+        to_emails=target.user_email,
+        subject=cfg["invitations.email_subject"],
+        html_content=(
+            f'{cfg["invitations.email_body_preamble"]}<br /><br />'
+            'Please click <a href="http://localhost:5000/login/google-oauth2/'
```

Comment:
This URL shouldn't be hardcoded. Perhaps needs to get pulled from the config file?

Reply:
`localhost` is hardcoded throughout; we don't have the `host` in the config yet, but I can add it and then change this. The rest of the app should be updated in another PR.